homework/promise.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+/* cSpell:disable */
+{
+  function fetchJSONWithPromise(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status < 400) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach(key => {
+      const value = options[key];
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: error.message, class: 'alert-error', role: 'alert' });
+  }
+
+  function app(repos) {
+    const root = document.getElementById('root');
+    const container = createAndAppend('div', root, {
+      class: 'container',
+    });
+    const header = createAndAppend('header', container, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    const select = createAndAppend('select', header, {
+      id: 'list',
+    });
+    for (let i = 0; i < repos.length; i++) {
+      createAndAppend('option', select, { text: repos[i].name, value: i });
+    }
+    const repoContainer = createAndAppend('div', container, { class: 'container-repo' });
+    const listInfo = createAndAppend('ul', repoContainer, { class: 'repo-info' });
+    const listContributor = createAndAppend('ul', repoContainer, { class: 'repo-contributor' });
+
+    // creating li elements to display repo info when repo name change (onchange event handler)
+    async function displayRepoInfo() {
+      removeChildren(listInfo);
+      const repoName = createAndAppend('li', listInfo);
+      repoName.innerHTML = `Repository: <a target="_blank" href= ${repos[select.value].html_url}>${
+        repos[select.value].name
+      }</a>`;
+      createAndAppend('li', listInfo, {
+        text: `Description: ${repos[select.value].description}`,
+      });
+      createAndAppend('li', listInfo, { text: `Forks: ${repos[select.value].forks}` });
+      createAndAppend('li', listInfo, {
+        text: `Updated: ${repos[select.value].updated_at}`,
```

Comment:
The `updated_at` date which is retrieved from the GitHub API comes in ISO 8601 format i.e 2018-04-06T17:18:06Z. This can be shown in a more readable format like `dd/mm/yyyy`.

For this, I would suggest that you explore the Date object of JavaScript and see if you can make this appear on UI in `dd/mm/yyyy` format - https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Date

Reply:
Thanks for the link! change is made! :) 👍